testsuite/performance/tests/sar.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+cd "$(dirname "$0")"
+. ./common.sh
+
+SAR_OPERATION=${SAR_OPERATION:-stop}
+
+SAR_HOME="$PROJECT_BUILD_DIRECTORY/sar"
+PID_FILE="$SAR_HOME/sar.pid"
+TIMESTAMP_FILE="$SAR_HOME/sar.timestamp"
+if [[ -f "$TIMESTAMP_FILE" ]]; then 
+    TIMESTAMP=`cat $TIMESTAMP_FILE`
+else
+    TIMESTAMP=`date +%s`
+fi
+SAR_RESULTS_HOME="$SAR_HOME/$TIMESTAMP"
+SAR_OUTPUT_FILE="$SAR_RESULTS_HOME/sar-output.bin"
+BUNZIP_OUTPUT_FILE=${BUNZIP_OUTPUT_FILE:-false}
+CPU_COUNT=${CPU_COUNT:-`grep -c ^processor /proc/cpuinfo`}
+
+case "$SAR_OPERATION" in
+
+    start)
+        if [[ ! -f "$PID_FILE" ]]; then
+            echo "Starting sar command."
+            mkdir -p $SAR_RESULTS_HOME
+            echo $TIMESTAMP > $TIMESTAMP_FILE
+            sar -A -o "$SAR_OUTPUT_FILE" 2 &>/dev/null & SAR_PID=$! && echo $SAR_PID > $PID_FILE
+        fi
+    ;;
+
+    stop)
+        if [[ -f "$PID_FILE" ]]; then
+            echo "Stopping sar command."
+            SAR_PID=`cat $PID_FILE`
+            kill $SAR_PID && rm $PID_FILE && rm $TIMESTAMP_FILE
+
+            GNUPLOT=${GNUPLOT:-false}
+            if $GNUPLOT ; then GNUPLOT_SCRIPT="$PROJECT_BASEDIR/src/main/gnuplot/sar.gnuplot"; fi
+
+            echo "Processing sar output. GNUPLOT: $GNUPLOT"
+
+            # ALL CPUS
+            sar -u -f $SAR_OUTPUT_FILE > $SAR_RESULTS_HOME/cpu-ALL.txt
+            sadf -d $SAR_OUTPUT_FILE > $SAR_RESULTS_HOME/cpu-ALL.csv
+            if $GNUPLOT ; then gnuplot -e "datafile='$SAR_RESULTS_HOME/cpu-ALL.csv'" $GNUPLOT_SCRIPT > "$SAR_RESULTS_HOME/cpu-ALL.png"; fi
+
+            # PER-CPU
+            for CPU in $(seq -f "%02g" 0 $(( CPU_COUNT-1 )) ); do 
+                sar -P $CPU -u -f $SAR_OUTPUT_FILE > $SAR_RESULTS_HOME/cpu-$CPU.txt
+                sadf -P $CPU -d $SAR_OUTPUT_FILE > $SAR_RESULTS_HOME/cpu-$CPU.csv
+                if $GNUPLOT ; then gnuplot -e "datafile='$SAR_RESULTS_HOME/cpu-$CPU.csv'" $GNUPLOT_SCRIPT > "$SAR_RESULTS_HOME/cpu-$CPU.png"; fi
+            done
+
+            if $BUNZIP_OUTPUT_FILE; then bzip2 "$SAR_OUTPUT_FILE"; fi
```

Comment:
Can you add:
- I/O stats
- memory use
- Context switch
- Network statistics

Reply:
Context switches and IO makes sense but since this is measured on the docker host the network stats and memory usage per container is not available. We would have to run sar inside the containers.

Reply:
ack, for now we can omit the mem and network